packages/ui/src/header/header.tsx

```diff
@@ -40,9 +43,22 @@ const LeftItem = (props: HeaderProps) => (
     {props.children}
   </div>
 );
-const RightItem = (props: HeaderProps) => (
-  <div className={cn("flex justify-end gap-3 w-auto items-baseline", props.className)}>{props.children}</div>
-);
+const RightItem = (props: HeaderProps) => {
+  const variant = React.useContext(HeaderContext);
```

Comment:
@gakshita Can you please add one check over here to throw an error if variant is `undefined`. In case where someone accidentally uses it outside the `HeaderContext` as it should always be inside it.